### DISCUSSION TITLE
Consider extending discard use to refs by allowing in-only value

### DISCUSSION THREAD

**Discussion by VWilcox2000** [dotnet/csharplang#6240] [Language Ideas] [answered]
I do not think this would be a high priority, but I wanted to throw it out there and didn't see any of the other discard threads referring to refs with an input-only value though #79 has discussion about intentionally not supporting ref _ to mean null input value.  I would like to do something like this:

          this.SimpleMethod(ref _ = true);  // send true in -- discard the output

For a typical use case, let's say I have a method, likely a recursion that uses a ref variable to track whether processing occurred already.

        private void SimpleMethod(ref bool processed)
        {
          ...
          processed = true;
          ...
        }

When I went to make a separate call to that method where the flag was inconsequential, I need to declare a ref and set it:

        private void Test()
        {
          bool processed;

          processed = true;
          this.SimpleMethod(ref processed);
        }

I found myself instinctively putting _ before realizing it won't work because this is a ref.

In discussion for #79, @333fred pointed out they avoided supporting ref _ (I believe in case of a nullable variable) because they reasonably felt that would allow coders to shoot themselves in the foot.

I agree with that, but I would like to specify explicitly an input-only value using 'in' or less likely to conflict is maybe something like this:

        private void Test()
        {
          this.SimpleMethod(ref _ = true);
        }

I don't use ref pattern all that often particularly because of inability to use in async methods, nor do I think this will make more than a small quality of life improvement, but wanted to throw it out there.

**Comment by svick** [comment, marked as answer]
This is a subset of https://github.com/dotnet/csharplang/issues/4355, so I'm marking this as duplicate.
